sqld/src/database/libsql.rs

```diff
@@ -264,11 +258,11 @@ impl LibSqlDb {
 
 #[async_trait::async_trait]
 impl Database for LibSqlDb {
-    async fn execute(&self, queries: Queries) -> anyhow::Result<(Vec<QueryResult>, State)> {
+    async fn execute(&self, queries: Queries) -> Result<(Vec<QueryResult>, State)> {
         let (resp, receiver) = oneshot::channel();
         let msg = Message { queries, resp };
         let _ = self.sender.send(msg);
 
-        Ok(receiver.await?)
+        receiver.await.map_err(Error::LibSqlReceiverError)
```

Comment:
I think Receive error can be collapsed to an internal error, type erased error. You can implement `From<tokio::RecvError> for Error` so you don't have to`map_err` and simply use `?`